core/src/main/java/org/polypheny/db/monitoring/events/BaseEvent.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2019-2021 The Polypheny Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.polypheny.db.monitoring.events;
+
+import java.sql.Timestamp;
+import java.util.UUID;
+import lombok.Getter;
+
+
+public abstract class BaseEvent implements MonitoringEvent {
+
+
+    @Getter
+    private final UUID id = UUID.randomUUID();
+    protected String eventType;
+
+
+    private long recordedTimestamp;
+
+
+    public BaseEvent() {
+        setEventType( eventType );
+        recordedTimestamp = getCurrentTimestamp();
+    }
+
+
+    public void setEventType( String eventType ) {
```

Comment:
@hennlo You once added this eventType? Should we remove it and just use the implementing class as type? I think I already removed some part of it

Reply:
This is still required, the implementing ClassType e.g. could be DML
The EventType is a String translation form SQLKind: INSERT/UPDATE/DELETE and is therefore more specific.

I would propose to keep the functionality

Reply:
But here we are not talking about sql or something like that. This is the base event of our monitoring.

If you want to have something like that for query events we can add it there, what do you think?

To be honest, I do not see any benefit of the string in the base event, we can always just get the implementing class. If a event need kind of type (which could even be different within the same class), i would implement it directly on this event and not on the base event.

Reply:
Ah okay, yes you're absolutely right, I looked at the wrong part!
Yes in that case eventType is only necessary in StataementEvent or Subsclasses.

I would propose StatementEvent.